### ISSUE TITLE
 paddle.nn.functional.dice_loss 计算出的值有点不对

### ISSUE THREAD

**Issue by DUT-LiMing** [PaddlePaddle/Paddle#32398]
- 标题： paddle.nn.functional.dice_loss 计算出的值有点不对
- 版本、环境信息：
    1）PaddlePaddle版本：2.0.2
    2）CPU：i7-8750H
    3）GPU：GTX1050TI
    4）系统环境：windows10   
  
      我在做一个分割任务时，用一张分割的不是很好的预测图和标签做dice计算，结果得到的指标很高，然后我自己按照公式写了一个计算dice的函数，得到的结果符合我的实际情况，下图第一个print 输出的是 用paddle.nn.functional.dice_loss 函数计算得到的结果，后两个print输出的都是我自己写的dice的结果。 
![JJVT6X`WFY3FSN}DS3J{761](https://user-images.githubusercontent.com/59254360/115444075-ad124c00-a246-11eb-9b82-a80e79321034.png)

![MW{WINUPPC6HIHPC HW%8Q](https://user-images.githubusercontent.com/59254360/115442241-5dcb1c00-a244-11eb-8c88-1d8efb227296.png)

 
![VIAMUX$HE92E78NO$LXOLCS](https://user-images.githubusercontent.com/59254360/115443137-77b92e80-a245-11eb-874a-32dcc7cb0ea3.png)

下图分别是预测图和标签图，可以看出分割的并不是很好 ，至少dice 指标应该不至于到达0.975
 
 
![ID7VJS0AC7ELB((E65U~{QL](https://user-images.githubusercontent.com/59254360/115443662-1b0a4380-a246-11eb-9aa9-d133ce50625f.png)

![P GWY8 9SS6`7BE%0TJH}0X](https://user-images.githubusercontent.com/59254360/115443685-22315180-a246-11eb-98e1-43454e0fac35.png)

**Comment by haoyuying** [comment]
您好同学，可否将上述两张计算dice_loss的图保存成png的0，1格式方便我们排查问题？感谢您的配合

**Comment by DUT-LiMing** [comment]
好的，图片在附件中，是用PIL保存的 0，1 的 图片 ，麻烦您了




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/Paddle"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月21日(星期三) 下午5:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/Paddle]  paddle.nn.functional.dice_loss 计算出的值有点不对 (#32398)


![predict](https://user-images.githubusercontent.com/59254360/115538215-3c167700-a2ce-11eb-9ae8-643a439c6e5b.png)
![label](https://user-images.githubusercontent.com/59254360/115538236-4173c180-a2ce-11eb-831e-8256fe7ac0fa.png)



 
您好同学，可否将上述两张计算dice_loss的图保存成png的0，1格式方便我们排查问题？感谢您的配合
